### PR TITLE
feat(client): Update session start options and project settings to use compute resource pools

### LINKED ISSUE #2484: Support Resource Pools from the UI
## Motivation

We are introducing [Resource Access Control](https://github.com/SwissDataScienceCenter/renku-design-docs/blob/main/rfcs/011-resource-access-control/011-resource-access-control.md) in RenkuLab to assign resource pools to specific users.

This requires a few changes in the UI since the session will start within the context of a specific resource pool with an assigned set of resources.

### Starting sessions

We will need to query an API to get the list of resources (it should be [the `GET /resource_pools` API here](https://github.com/SwissDataScienceCenter/renku-data-services/blob/main/src/api.spec.yaml)) available to the user.
We will also need to query the project options to get a project's default pool and try to set it.
For quick-started sessions, we will use whatever is the project default or just the global default if nothing local overwrites it; for sessions through the "Start with options" page, we will error if the project-defined pool is not available, and show a warning. User can pick any other available option from the pool on that page.

ℹ️ This means the single resources (CPU, RAM, disk, GPU) won't be selectable as single entities but just as a pre-set. We won't need the `notebook_options` anymore since there will be just pools with sets of resources.

image

### Sessions settings
On the project page, users can set session settings. The abovementioned ones will disappear and be swapped with the new pool sets.
Keep in mind that not everything is disappearing from there. The pinned images won't change, nor the default_url (even tho it will be just a pure string), and _probably_ also the "Automatically fetch LFS data" should not change.

### PR DESCRIPTION
**Pull request by leafty**
Feature branch collecting changes to support the new compute resource pools for sessions.

Fixes #2484.

Related: https://github.com/SwissDataScienceCenter/renku/pull/3067

/deploy renku=crac-service-helm-chart renku-notebooks=master #persist #cypress

### CODE REVIEW

File: client/src/features/project/projectCoreApi.ts [hunk 1]
```diff
@@ -178,11 +236,161 @@ export const projectCoreApi = createApi({
         { type: "project-status", id: migrationParams.gitUrl },
       ],
     }),
+    getConfig: builder.query<ProjectConfig, GetConfigParams>({
+      query: ({ projectRepositoryUrl, versionUrl }) => {
+        const params = {
+          git_url: projectRepositoryUrl,
+          // Branch option not working currently
+          // ...(branch ? { branch } : {}),
```
Comment (lorenzo-cavazzi) [inline client/src/features/project/projectCoreApi.ts:244]: :scream: 
Is there an issue on renku-python we can add for reference here?

File: client/src/features/session/startSessionOptions.d.ts
Comment (lorenzo-cavazzi) [inline client/src/features/session/startSessionOptions.d.ts:1]: Should we rename it as `session.types.ts` as discussed for the coding style?

File: client/src/features/dataServices/dataServices.d.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  /** Max disk storage in Gigabytes */
+  max_storage: number;
+
+  /** Default disk storage in Gigabytes */
+  default_storage: number;
+
+  default: boolean;
+
+  // eslint-disable-next-line spellcheck/spell-checker
```
Comment (leafty) [inline client/src/features/dataServices/dataServices.d.ts:48]:
```suggestion
```

File: client/src/features/dataServices/dataServices.d.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ * limitations under the License.
+ */
+
+export interface ResourcePool {
+  id: number;
+  name: string;
+  classes: ResourceClass[];
+  quota: Resources;
+  // eslint-disable-next-line spellcheck/spell-checker
+  // TODO(@leafty): remove "?"
```
[the remark this answers is not in the document]
Comment (leafty) [reply client/src/features/dataServices/dataServices.d.ts:25]:
```suggestion
```

File: client/src/features/dataServices/dataServices.d.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  default?: boolean;
```
Comment (leafty) [inline client/src/features/dataServices/dataServices.d.ts:26]:
```suggestion
  default: boolean;
```

File: client/src/features/dataServices/dataServices.d.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  public?: boolean;
```
Comment (leafty) [inline client/src/features/dataServices/dataServices.d.ts:27]:
```suggestion
  public: boolean;
```

File: client/src/features/dataServices/dataServices.d.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface ResourcePool {
+  id: number;
+  name: string;
+  classes: ResourceClass[];
+  quota: Resources;
+  // eslint-disable-next-line spellcheck/spell-checker
```
Comment (leafty) [inline client/src/features/dataServices/dataServices.d.ts:24]:
```suggestion
```

File: client/src/features/dataServices/dataServices.d.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  /** Max disk storage in Gigabytes */
+  max_storage: number;
+
+  /** Default disk storage in Gigabytes */
+  default_storage: number;
+
+  default: boolean;
+
+  // eslint-disable-next-line spellcheck/spell-checker
+  // TODO(@leafty): remove "?"
```
[the remark this answers is not in the document]
Comment (leafty) [reply client/src/features/dataServices/dataServices.d.ts:49]:
```suggestion
```

File: client/src/features/dataServices/dataServices.d.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  matching?: boolean;
```
Comment (leafty) [inline client/src/features/dataServices/dataServices.d.ts:50]:
```suggestion
  matching: boolean;
```

File: client/src/features/session/startSessionOptions.d.ts
Comment (leafty) [reply client/src/features/session/startSessionOptions.d.ts:1]: Will rename it as `startSessionOption.types.ts`. Maybe I will merge it with session types with the persistence refactor.

File: client/src/features/project/projectCoreApi.ts (hunk 1 above)
Comment (leafty) [reply client/src/features/project/projectCoreApi.ts:244]: Ah, actually, this should work now even if we will not use it (not sure how the UI should handle it).

Comment (lorenzo-cavazzi) [reply client/src/features/project/projectCoreApi.ts:244]: Yeah we currently don't allow to pick a different branch for interacting with the repo, but that was discussed a while ago and should happen at some point (currently low priority)

Comment (leafty) [reply client/src/features/project/projectCoreApi.ts]: Done in ab0fc30.

**Pull request merged**